packages/firestore/lite/src/api/reference.ts

```diff
@@ -533,6 +556,38 @@ export function addDoc<T>(
     .then(() => docRef);
 }
 
+export function refEqual<T>(
+  left: firestore.DocumentReference<T> | firestore.CollectionReference<T>,
+  right: firestore.DocumentReference<T> | firestore.CollectionReference<T>
+): boolean {
+  if (
```

Comment:
is this if() just to guard against invalid types from JS?

Reply:
It also does the type casting which exposes the private API.